server/src/main.rs

```diff
@@ -43,10 +51,22 @@ pub struct AppState {
 fn run_server(host: &str, port: &str, repo_root: &Path) {
     let _sys = System::new("gitkv-server");
 
-    let addr = GitRepos::new(git::load_repos(&repo_root)).start();
+    let repos = git::load_repos(&repo_root);
 
+    if log_enabled!(Level::Info) {
```

Comment:
You can actually avoid this entire `if` and all the prep for formatting the `Vec` here by using the debug formatter:

```suggestion
    info!("Loaded Git repos: [{:?}]", repos.keys());
```
You can find all the nice little format strings you can use instead of just `{}` documented in `std::fmt` at https://doc.rust-lang.org/std/fmt/index.html.

I also popped together an example for you to prove it works:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=9e583fc2b199c1174a6c820c2395563d

This outputs:

> Loaded Git repos: ["yet-another-repo", "some-repo", "another-repo"]

Which is close enough to the original and yet does absolutely 0 memory allocation so bonus!

Reply:
Much cleaner 👍